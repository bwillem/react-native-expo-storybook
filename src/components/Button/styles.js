import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderRadius: 10,
    borderWidth: 1,
    paddingVertical: '1%',
    paddingHorizontal: '3%',
    marginVertical: '3%',
    marginHorizontal: '5%',
    alignItems: 'center',
    justifyContent: 'center'
  },
  text: {
    alignSelf: 'center'
  }
});
