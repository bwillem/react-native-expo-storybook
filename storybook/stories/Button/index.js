import React from 'react';
import { View } from 'react-native';

import { storiesOf } from '@storybook/react-native';
import { select, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import { Button } from '@components';
import { colors, fontSize } from '@themes';

import { iconOptions } from 'ExpoStorybook/src/components/Icon/constants';

const sizeCapOptions = Object.keys(fontSize);
const colorOptions = Object.keys(colors);

storiesOf('Button', module)
  .add('custom', () => (
    <Button
      onPress={action('pressed-custom')}
      color={select('Color', colorOptions, colorOptions[0])}
      colorText={select('colorText', colorOptions, colorOptions[1])}
      size={select('Size', sizeCapOptions, sizeCapOptions[1])}
      value={text('Value', 'Hello')}
    />
  ))
  .add('multiple', () => (
    <View>
      <Button
        onPress={action('pressed-1')}
        value={text('Example1', 'Button')}
      />
      <Button
        iconName="pencil"
        onPress={action('pressed-2')}
        value={text('Example2', 'Edit')}
      />
      <Button
        iconName={select('IconName', iconOptions, iconOptions[0])}
        onPress={action('pressed-3')}
        value={text('Example3', 'Custom icon')}
      />
    </View>
  ));
