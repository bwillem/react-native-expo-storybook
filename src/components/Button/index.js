import React from 'react';
import PropTypes from 'prop-types';
import { Text, TouchableOpacity, ViewPropTypes } from 'react-native';

import { Icon } from '@components';
import { colors, fontSize } from '@themes';

import styles from './styles';

const Button = props => {
  const { style, onPress, iconName, value, color, colorText, size } = props;

  const stylesContainer = [
    {
      backgroundColor: colors[color]
    },
    styles.container,
    style
  ];
  const stylesText = [
    {
      fontSize: fontSize[size],
      color: colors[colorText]
    },
    styles.text
  ];

  return (
    <TouchableOpacity onPress={onPress} style={stylesContainer}>
      {iconName && <Icon name={iconName} color={colorText} />}
      <Text style={stylesText}>{value}</Text>
    </TouchableOpacity>
  );
};

Button.propTypes = {
  size: PropTypes.oneOf(Object.keys(fontSize)).isRequired,
  color: PropTypes.oneOf(Object.keys(colors)).isRequired,
  colorText: PropTypes.oneOf(Object.keys(colors)).isRequired,
  value: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  style: ViewPropTypes.style,
  iconName: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
};
Button.defaultProps = {
  iconName: false,
  color: 'primary',
  colorText: 'white',
  style: {},
  size: 'medium'
};

export default Button;
