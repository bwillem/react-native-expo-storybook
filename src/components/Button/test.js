import { shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';

import Button from './';

describe('Button testing renders', () => {
  it('with text', () => {
    const wrapper = shallow(<Button value="Text" onPress={() => undefined} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('with color prop', () => {
    const wrapper = shallow(
      <Button color="secondary" value="Text color" onPress={() => undefined} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('with iconName prop', () => {
    const wrapper = shallow(
      <Button iconName="pencil" value="Text icon" onPress={() => undefined} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ButtonText testing events', () => {
  it('should handle button pressed', () => {
    const onPress = sinon.spy();
    const wrapper = shallow(<Button value="onPress" onPress={onPress} />);

    wrapper.simulate('press');
    expect(onPress).toHaveProperty('callCount', 1);
  });
});
