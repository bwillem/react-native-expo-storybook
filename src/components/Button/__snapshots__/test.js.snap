// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button testing renders with color prop 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "tomato",
      },
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": "5%",
        "marginVertical": "3%",
        "paddingHorizontal": "3%",
        "paddingVertical": "1%",
      },
      Object {},
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 40,
        },
        Object {
          "alignSelf": "center",
        },
      ]
    }
  >
    Text color
  </Text>
</TouchableOpacity>
`;

exports[`Button testing renders with iconName prop 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "darkslateblue",
      },
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": "5%",
        "marginVertical": "3%",
        "paddingHorizontal": "3%",
        "paddingVertical": "1%",
      },
      Object {},
    ]
  }
>
  <Icon
    color="white"
    name="pencil"
    size={50}
    style={Object {}}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 40,
        },
        Object {
          "alignSelf": "center",
        },
      ]
    }
  >
    Text icon
  </Text>
</TouchableOpacity>
`;

exports[`Button testing renders with text 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "darkslateblue",
      },
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": "5%",
        "marginVertical": "3%",
        "paddingHorizontal": "3%",
        "paddingVertical": "1%",
      },
      Object {},
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 40,
        },
        Object {
          "alignSelf": "center",
        },
      ]
    }
  >
    Text
  </Text>
</TouchableOpacity>
`;
